### DISCUSSION TITLE
rpmsg_ns_bind_cb context?

### DISCUSSION THREAD

**Discussion by infn-ke** [OpenAMP/open-amp#273] [Q&A] [answered]
When using API `rpmsg_init_vdev` I need a context passed to the name service callback `rpmsg_ns_bind_cb`. How can that be achieved?

**Comment by arnopo** [comment, marked as answer]
Hello @infn-ke 

It seems more a question than an issue. i convert it to discussions
Please, next time, open a discussion instead of an issue, for questions.

To retrieve your context you could implement something like the following.
```
struct my_ctx_struct {
	struct rpmsg_virtio_device rpmsg_vdev;
};
```
then in you init code
```
struct my_ctx_struct *my_ctx;

my_ctx = metal_allocate_memory(sizeof( *my_ctx));
rpmsg_init_vdev (my_ctx->rpmsg_vdev, ...);
```
In your  rpmsg_ns_bind_cb callback 
```
 struct my_ctx_struct *my_ctx;
 my_ctx  =    metal_container_of(rpmsg_vdev, struct my_ctx_struct, rpmsg_vdev); 
```
